### ISSUE TITLE
Voiceover translation of the video

### ISSUE THREAD

**Issue by myrky** [TeamNewPipe/NewPipe#10393]
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I'm aware that this is a request for NewPipe itself and that requests for adding a new service need to be made at [NewPipeExtractor](https://github.com/TeamNewPipe/NewPipeExtractor/issues).
- [X] I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Feature description

On GitHub, there is a browser extension available that adds voiceover translation to videos. Here's the link: https://github.com/ilyhalight/voice-over-translation/blob/master/README-EN.md. It would be incredibly cool if this script could be integrated into your application.

### Why do you want this feature?

Integrating the voiceover translation script would make your application more user-friendly and engaging. This feature would allow users to watch videos with synchronized voiceover translations in their preferred language. As a result, the content would become accessible to a broader audience, including individuals who are not native speakers of the language.

**Comment by myrky** [comment]
> The only website we support on that list is Youtube. Since YT itself is adding support for audio in multiple languages, I wonder how useful this will be. Currently, it's up to the creators to do this, but it's pretty easy to extrapolate that eventually YT will use AI to provide translated audio in other languages.

I think it'll be useful. I've seen various people start asking for this script to be implemented on the Revansed forum a long time ago, but apparently the developers of this application are immune to requests.

Personally, I would enjoy watching videos in Korean, Japanese and other languages without waiting for someone to translate.

**Comment by myrky** [comment]
> But the readme says it only has translation to Russian available from other languages? And the only language for which English translations are available is Russian?

No, there are many different languages in there, both the target language and the source language. I suggest you install it in your browser and see how it works :)

**Comment by Stypox** [comment]
Closing as now YouTube itself has voiceovers for most videos. Furthermore, the github project you linked sends audio data to Yandex, which is something we definitely won't implement in NewPipe as a privacy app. Only a local machine learning model would be acceptable privacy-wise, but then we'd have to make the APK much bigger to fit the ML libraries, so this won't be added to newpipe.

**Comment by vukitoso** [comment]
> Furthermore, the github project you linked sends audio data to Yandex, which is something we definitely won't implement in NewPipe as a privacy app.

@Stypox, what exactly is a privacy violation?
You do not send your personal data.
This translation function will only be used after an explicit button click. By default there is no translation. And therefore no additional requests are initiated anywhere.

I don't see any breach of confidentiality.

If by violation of confidentiality you understand the fact that Yandex can link your IP address with some videos that you requested for translation, then, as I already said, the video translation function will be available only after you explicitly click on the “translate video through Yandex” button.

I think this is 100% safe functionality.
